TestProjects/ShaderFoundry/Assets/CommonAssets/Editor/EditorTests/BlockLibrary/VirtualTexturePropertyBlockBuilder.cs

```diff
@@ -0,0 +1,153 @@
+using System.Collections.Generic;
+using UnityEditor.ShaderFoundry;
+using static UnityEditor.ShaderFoundry.VirtualTextureLayerAttribute;
+
+namespace UnityEditor.ShaderFoundry.UnitTests
+{
+    internal class VirtualTexturePropertyBlockBuilder : BasePropertyBlockBuilder
+    {
+        public class LayerData
+        {
+            internal string UniformName;
+            internal string DisplayName;
+            internal string TextureName;
+            internal LayerTextureType? TextureType = null;
+        }
+
+        const string VirtualTexturePropertyTypeName = "VTPropertyWithTextureType";
+        public int LayerCount = 2;
+        public int LayerToSample = 0;
+        
+        public List<LayerData> Layers = new List<LayerData>
+        {
+            new LayerData(),
+            new LayerData(),
+            new LayerData(),
+            new LayerData(),
+        };
+
+        public static string GetName(int layerIndex, string layerValue, string propertyValue, string fieldName)
+        {
+            // Uniform and Display names follow a fallback scheme. It will use in order:
```

Comment:
I don't see any OOB checks here for layerIndex - is this intended?

Reply:
I guess I didn't think about it, but I believe this would just cause the test to fail which would be fine. An OOB check would potentially give better error messaging, but this should be caught at test authoring time.